File: java/1022/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumRootToLeaf(TreeNode root) {
        return sumNodeToLeaf(root, 0);
    }

    private int sumNodeToLeaf(TreeNode start, int current) {
        if (start == null) {
            return 0;
        }

        int answer = (current << 1) + start.val;

        if (start.left == null
           && start.right == null) {
            return answer;
        }

        return sumNodeToLeaf(start.left, answer) + sumNodeToLeaf(start.right, answer);
    }
}

File: python3/1022/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode]) -> int:
        return self.sumNodeToLeaf(root, 0)

    def sumNodeToLeaf(self, start: Optional[TreeNode], current: int) -> int:
        if start == None:
            return 0

        answer = (current << 1) + start.val

        if start.left == None and start.right == None:
            return answer

        return self.sumNodeToLeaf(start.left, answer) + self.sumNodeToLeaf(start.right, answer)

